### ISSUE TITLE
power usage does not include drain

### ISSUE THREAD

**Issue by taleden** [KirkMcDonald/kirkmcdonald.github.io#220]
The "power" column seems to only reflect the active power use of the factories, but not their passive power drain. Per the [wiki](https://wiki.factorio.com/Electric_system#Consumption): "Drain is cumulative with energy cosumption - for example, an active Assembling machine 2 will consume 155 kW (150kW energy consumption + 5kW drain)."

To reproduce, configure [a single Assembling machine 1 producing anything](https://kirkmcdonald.github.io/calc.html#data=1-0-0&items=wooden-chest:f:1) and note its "power" value: 75kW. Then look up [that same factory in the wiki](https://wiki.factorio.com/Assembling_machine_1) and note that its (active) "energy consumption" is 75kW, but it also has a "drain" of 2.5kW that the calculator omits.

**Comment by DRY411S** [comment]
Ok, but why does this matter? Personally I use the kirkmcdonald to estimate the maximum power that I must supply. If I build enough power to support the maximum, by default, I support the drain.

**Comment by taleden** [comment]
No, that's the point: the power totals shown on the calculator do not include the drain, so if you only provide exactly that much power, the factories will be underpowered in game because they require their drain power on top of their active power.

**Comment by DRY411S** [comment]
I don't believe that's true. Drain is separate to when the machine works, not additive.

**Comment by taleden** [comment]
Read again the statement on the wiki that I linked and also quoted in the report: "Drain is **cumulative** with energy consumption - for example, an active Assembling machine 2 will consume 155 kW (150kW energy consumption **+** 5kW drain)."

To confirm, here's a screenshot of an actual Assembling machine 2 in-game:
![am2](https://user-images.githubusercontent.com/23055012/98254642-28ef3500-1f42-11eb-8aee-b83433766bcc.png)
Note that it says "Max. consumption: 155 kW". That's the 150kW active consumption **plus** the 5 kW drain **in addition**.

**Comment by DRY411S** [comment]
Happy to stand corrected. Definitely a bug the Devs need to address then.

**Comment by taleden** [comment]
I am new to the game, so it's very possible that drain was separate at some point in the past and that assumption continues to propagate by outdated guides and such. :)

**Comment by taleden** [comment]
On the topic of power calculations, I'll also take the opportunity to bump issue #113: when a dependency is clicked to ignore it, there should be an option to also exclude that dependency's power requirement from the "total power" tally. It seems to me the point of ignoring dependencies is to indicate that they're produced elsewhere, which means not only their sub-dependencies but also their direct power dependency is not relevant to the current design and should be excluded from the power total.

**Comment by DRY411S** [comment]
> I am new to the game, so it's very possible that drain was separate at some point in the past and that assumption continues to propagate by outdated guides and such. :)

or other people not reading things carefully enough, when they are directed to the exact information. 👍

**Comment by taleden** [comment]
Looks like PR #191 already addresses this actually, although it calculates drain as 1/30th of active power which may or may not be true for all entities; it'd be better to add drain to the data definitions from the game.
